packages/react/src/components/SidePanel/index.js

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+
+import Heading from '../Heading';
+import IconButton from '../IconButton';
+import Overlay from '../Overlay';
+
+const SidePanel = ({
+  children,
+  className,
+  footer,
+  isOpen: isOpenProp,
+  onClose,
+  title,
+  ...props
+}) => {
+  const [isOpen, setIsOpen] = React.useState(isOpenProp);
+
+  const toggleIsOpen = () => setIsOpen(!isOpen);
+
+  const handleClose = () => {
+    toggleIsOpen();
+    onClose();
+  };
+
+  React.useEffect(() => {
+    setIsOpen(isOpenProp);
+  }, [isOpenProp]);
+
+  return (
+    <>
+      <Overlay
+        className={cx({
+          'is-hidden': !isOpen,
+        })}
+        onClick={handleClose}
```

Comment:
🌷 How about a prop `overlayClickClosesPanel` set default to `true`?

Reply:
I'd rather go with `should` or `close` at the start because if we want to add other conditions (like close the panel when pressing ESC), it should have the same pattern.
So what about
`shouldCloseOnOverlayClick`
`closesWithOverlay`
`closesOnOverlayClick`
?

Reply:
I like `closesOnOverlayClick`

Reply:
✅